packages/transformers/vue/src/VueTransformer.js

```diff
@@ -64,10 +64,24 @@ export default (new Transformer({
       });
     }
 
+    const descriptor = parsed.descriptor;
+    let id = hashObject({
+      filePath: asset.filePath,
+    }).slice(-6);
```

Comment:
This seems a bit sketchy. Why does this ID only depend upon the asset's location?

Reply:
[`vue-loader`](https://github.com/vuejs/vue-loader/blob/e9314347d75a1b0e54f971272d23a669fc3e6965/src/index.ts#L117-L121) and [Vite](https://github.com/vitejs/vite/blob/99740949b77509c442a27fb0e017fcbe376f3391/packages/plugin-vue/src/utils/descriptorCache.ts#L30) both use the source code for the Vue component as well in production. We should probably do the same. You can add this ID to the returned descriptor:

```js
let id = hashObject({
  filePath: asset.filePath,
  source: options.mode === 'production' ? code : null
});

return {
  type: 'vue',
  version: '3.0.0',
  program: {
    ...,
    id
  },
}
```

Later, when we generate the same ID at the top of `transform`, you can replace it with:

```js
let {template, script, styles, customBlocks, id} = nullthrows(
  await asset.getAST(),
).program;
```

Reply:
Ok, changed and pushed!